pkg/statdb/statdb.go

```diff
@@ -5,428 +5,145 @@ package statdb
-		dbx.Node_UptimeSuccessCount(uptimeSuccessCount),
-		dbx.Node_TotalUptimeCount(totalUptimeCount),
-		dbx.Node_UptimeRatio(uptimeRatio),
-	)
-	if err != nil {
-		return nil, Error.Wrap(status.Errorf(codes.Internal, err.Error()))
-	}
+	// UpdateBatch for updating multiple farmers' stats in the db
+	UpdateBatch(ctx context.Context, updateBatchReq *UpdateBatchRequest) (resp *UpdateBatchResponse, err error)
 
-	nodeStats := &pb.NodeStats{
-		NodeId:            node.Id,
-		AuditSuccessRatio: dbNode.AuditSuccessRatio,
-		AuditCount:        dbNode.TotalAuditCount,
-		UptimeRatio:       dbNode.UptimeRatio,
-		UptimeCount:       dbNode.TotalUptimeCount,
-	}
-	return &pb.CreateResponse{
-		Stats: nodeStats,
-	}, nil
+	// CreateEntryIfNotExists creates a statdb node entry and saves to statdb if it didn't already exist
+	CreateEntryIfNotExists(ctx context.Context, createIfReq *CreateEntryIfNotExistsRequest) (resp *CreateEntryIfNotExistsResponse, err error)
 }
 
-// Get a storagenode's stats from the db
-func (s *StatDB) Get(ctx context.Context, getReq *pb.GetRequest) (resp *pb.GetResponse, err error) {
-	defer mon.Task()(&ctx)(&err)
-
-	dbNode, err := s.DB.Get_Node_By_Id(ctx, dbx.Node_Id(getReq.NodeId.Bytes()))
-	if err != nil {
-		return nil, Error.Wrap(status.Errorf(codes.Internal, err.Error()))
-	}
-
-	nodeStats := &pb.NodeStats{
-		NodeId:            getReq.NodeId,
-		AuditSuccessRatio: dbNode.AuditSuccessRatio,
-		AuditCount:        dbNode.TotalAuditCount,
-		UptimeRatio:       dbNode.UptimeRatio,
-		UptimeCount:       dbNode.TotalUptimeCount,
-	}
-	return &pb.GetResponse{
-		Stats: nodeStats,
-	}, nil
+// CreateRequest is a statdb create request message
+type CreateRequest struct {
+	Node  storj.NodeID
+	Stats *pb.NodeStats
 }
 
-// FindInvalidNodes finds a subset of storagenodes that fail to meet minimum reputation requirements
-func (s *StatDB) FindInvalidNodes(ctx context.Context, getReq *pb.FindInvalidNodesRequest) (resp *pb.FindInvalidNodesResponse, err error) {
-	defer mon.Task()(&ctx)(&err)
-
-	var invalidIds storj.NodeIDList
-
-	nodeIds := getReq.NodeIds
-	maxAuditSuccess := getReq.MaxStats.AuditSuccessRatio
-	maxUptime := getReq.MaxStats.UptimeRatio
-
-	rows, err := s.findInvalidNodesQuery(nodeIds, maxAuditSuccess, maxUptime)
-
-	if err != nil {
-		return nil, Error.Wrap(err)
-	}
-	defer func() {
-		err = rows.Close()
-		if err != nil {
-			s.log.Error(err.Error())
-		}
-	}()
-
-	for rows.Next() {
-		node := &dbx.Node{}
-		err = rows.Scan(&node.Id, &node.TotalAuditCount, &node.TotalUptimeCount, &node.AuditSuccessRatio, &node.UptimeRatio)
-		if err != nil {
-			return nil, Error.Wrap(err)
-		}
-		id, err := storj.NodeIDFromBytes(node.Id)
-		if err != nil {
-			return nil, Error.Wrap(err)
-		}
-		invalidIds = append(invalidIds, id)
-	}
-
-	return &pb.FindInvalidNodesResponse{
-		InvalidIds: invalidIds,
-	}, nil
+// CreateResponse is a statdb create response message
+type CreateResponse struct {
+	Stats *pb.NodeStats
```

Comment:
Return the actual results rather than some wrapper.

Reply:
moby suggested he will work on refactoring all the structures